### ISSUE TITLE
Favorites in main menu not working

### ISSUE THREAD

**Issue by timmersr** [nextcloud/android#1585]
### Actual behaviour
On selection of Favorites from the menu "Loading..." is displayed

### Expected behaviour
Display as Favorite marked folders and files.
 
### Steps to reproduce
1. select Favorites from the menu.
2. 
3. 

### Environment data
Android version: 7.0
Device model: Samsung Galaxy 6 Edge Plus
Stock or customized system: Stock
Nextcloud app version: 2.0.0RC7
Nextcloud server version: 12.0.3

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by mario** [comment]
Can you please provide the logs?

**Comment by j-ed** [comment]
@mario I just realized the same problem. Some files and folders which have been marked as "Favorite" are accessible without any problems using the normal directory structure but if I try to open them using the Favorites menu, the app is running in a timeout.
What log file do you exactly require? The web server log and Nextcloud server log are not showing any related information.

**Comment by timmersr** [comment]
@mario Nothing in the server logs.

**Comment by AndyScherzinger** [comment]
@j-ed @mario does the list stay empty or does it "just" show the loading spinner? Just asking since depending on the amount of items to be found it might take quite a while to fetch&display the information, especially for account with large amounts on images/favorites.

**Comment by j-ed** [comment]
@AndyScherzinger This is my directory layout:
```
All files (3 files, 11 folders)
- folder-1 (7 folders) < 1s
  - folder-2 (2 files, 9 folders) < 1s
    - folder-3 (19 folders (4 starred)) < 1s
      - folder-4 (starred) (1 file) < 1s
```
If I select the `Favorites` menu entry it last approximately 9s to get "8 folders" folders displayed, **although only 4 folders** have been starred. This means all folders are displayed twice!
Then I select the starred **folder-4** and the spinning wheel is displayed. After 90s I decided to press the back button to return to the Favorites but what a surprise, the content of **folder-3** is displayed instead. If I select **folder-4** now its content (1 file) is instantly shown.

**Comment by AndyScherzinger** [comment]
Pinging @mario @tobiasKaminsky

**Comment by tobiasKaminsky** [comment]
Related to #1732

**Comment by tobiasKaminsky** [comment]
Fixed by PRs mentioned in #1732
